### ISSUE TITLE
Failed to load NER model

### ISSUE THREAD

**Issue by Julm** [ICIJ/datashare#626]
**Describe the bug**
Datashare failed to extract named entities because it could not load NER model

**To Reproduce**
0. Put a document in the data dir
1. Go to 'Analyse your documents'
2. Click on 'Extract text'
3. Click on 'Find people...'
4. See error:
```
2020-09-09 10:49:06,871 [Worker: RequestDispatcher: Thread-32] INFO  AbstractModels - synchronize models is set to true
2020-09-09 10:49:06,873 [Worker: RequestDispatcher: Thread-32] INFO  TaskResource - waiting for NlpApp org.icij.datashare.text.nlp.corenlp.CorenlpPipeline@7f8edf48 to listen...
2020-09-09 10:49:06,874 [pool-2-thread-3] INFO  NlpApp - running NlpApp for CORENLP pipeline with 1 thread(s)
2020-09-09 10:49:06,879 [Worker: RequestDispatcher: Thread-32] INFO  TaskResource - ...org.icij.datashare.text.nlp.corenlp.CorenlpPipeline@7f8edf48 is listening
2020-09-09 10:49:06,880 [pool-2-thread-4] INFO  ResumeNlpTask - resuming NLP name finding for index local-datashare and [CORENLP]
2020-09-09 10:49:06,959 [pool-2-thread-4] INFO  ResumeNlpTask - sent EXTRACT_NLP message for 1 files without [CORENLP] pipeline tags
2020-09-09 10:49:06,965 [CORENLP-0] INFO  NlpConsumer - extracting CORENLP entities for document 446bd81c5cb723cf0e2549be9dad347997e23495d0831518684170b42713b74ed165479f409f6542ccbe73a3fa43aebb
2020-09-09 10:49:06,966 [pool-2-thread-3] INFO  MemoryDataBus - exiting org.icij.datashare.nlp.NlpForwarder$$Lambda$2423/35493809@56897987
2020-09-09 10:49:06,966 [pool-2-thread-3] INFO  NlpApp - forwarder exited waiting for consumer(s) to finish
2020-09-09 10:49:06,967 [CORENLP-0] INFO  CorenlpPipeline - initializing CORENLP FRENCH [TOKEN, SENTENCE, POS, LEMMA, NER]
2020-09-09 10:49:07,150 [CORENLP-0] INFO  CoreNlpNerModels - downloading models for language FRENCH under dist/models/corenlp/4-0-0/fr
2020-09-09 10:49:17,387 [CORENLP-0] INFO  CoreNlpNerModels - models successfully downloaded for language FRENCH
2020-09-09 10:49:17,390 [CORENLP-0] INFO  CoreNlpNerModels - downloading models for language ENGLISH under dist/models/corenlp/4-0-0/en
2020-09-09 10:49:32,841 [elasticsearch[jVsOze3][management][T#3]] WARN  DiskThresholdMonitor - flood stage disk watermark [95%] exceeded on [jVsOze3SRVS1utr1n8VuHg][jVsOze3][C:\Users\julien.martin\AppData\Roaming\Datashare\index\nodes\0] free: 4.8gb[2%], all indices on this node will be marked read-only
2020-09-09 10:49:38,931 [CORENLP-0] INFO  CoreNlpNerModels - models successfully downloaded for language ENGLISH
2020-09-09 10:49:38,932 [CORENLP-0] INFO  CoreNlpNerModels - adding models\corenlp\4-0-0\en\stanford-corenlp-4.0.0-models-en.jar to system classloader
2020-09-09 10:49:39,004 [CORENLP-0] ERROR CoreNlpNerModels - failed loading NER
java.io.IOException: Unable to open "edu\stanford\nlp\models\ner\english.all.3class.caseless.distsim.crf.ser.gz" as class path, filename or URL
        at edu.stanford.nlp.io.IOUtils.getInputStreamFromURLOrClasspathOrFileSystem(IOUtils.java:481)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1505)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1497)
        at edu.stanford.nlp.ie.crf.CRFClassifier.getClassifier(CRFClassifier.java:2888)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:75)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)


        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:68)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)
        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.initializeNerAnnotator(CorenlpPipeline.java:179)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.initialize(CorenlpPipeline.java:79)
        at org.icij.datashare.nlp.NlpConsumer.findNamedEntities(NlpConsumer.java:72)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:43)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:20)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2020-09-09 10:49:39,005 [CORENLP-0] INFO  CoreNlpNerModels - adding models\corenlp\4-0-0\fr\stanford-corenlp-4.0.0-models-fr.jar to system classloader
2020-09-09 10:49:39,012 [CORENLP-0] ERROR CoreNlpNerModels - failed loading NER
java.io.IOException: Unable to open "edu\stanford\nlp\models\ner\french-wikiner-4class.crf.ser.gz" as class path, filename or URL
        at edu.stanford.nlp.io.IOUtils.getInputStreamFromURLOrClasspathOrFileSystem(IOUtils.java:481)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1505)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1497)
        at edu.stanford.nlp.ie.crf.CRFClassifier.getClassifier(CRFClassifier.java:2888)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:75)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)
        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.initializeNerAnnotator(CorenlpPipeline.java:179)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.initialize(CorenlpPipeline.java:79)
        at org.icij.datashare.nlp.NlpConsumer.findNamedEntities(NlpConsumer.java:72)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:43)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:20)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2020-09-09 10:49:39,013 [CORENLP-0] INFO  CorenlpPipeline - name-finding for FRENCH
2020-09-09 10:49:39,052 [CORENLP-0] INFO  CoreNlpNerModels - downloading models for language FRENCH under dist/models/corenlp/4-0-0/fr
2020-09-09 10:49:44,095 [CORENLP-0] INFO  CoreNlpNerModels - models successfully downloaded for language FRENCH
2020-09-09 10:49:44,138 [CORENLP-0] INFO  CoreNlpNerModels - downloading models for language ENGLISH under dist/models/corenlp/4-0-0/en
2020-09-09 10:49:56,349 [CORENLP-0] INFO  CoreNlpNerModels - models successfully downloaded for language ENGLISH
2020-09-09 10:49:56,350 [CORENLP-0] INFO  CoreNlpNerModels - adding models\corenlp\4-0-0\en\stanford-corenlp-4.0.0-models-en.jar to system classloader
2020-09-09 10:49:56,351 [CORENLP-0] ERROR CoreNlpNerModels - failed loading NER
java.io.IOException: Unable to open "edu\stanford\nlp\models\ner\english.all.3class.caseless.distsim.crf.ser.gz" as class path, filename or URL
        at edu.stanford.nlp.io.IOUtils.getInputStreamFromURLOrClasspathOrFileSystem(IOUtils.java:481)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1505)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1497)
        at edu.stanford.nlp.ie.crf.CRFClassifier.getClassifier(CRFClassifier.java:2888)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:75)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)
        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:68)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)
        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.processNerClassifier(CorenlpPipeline.java:196)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.process(CorenlpPipeline.java:91)
        at org.icij.datashare.nlp.NlpConsumer.findNamedEntities(NlpConsumer.java:73)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:43)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:20)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2020-09-09 10:49:56,352 [CORENLP-0] INFO  CoreNlpNerModels - adding models\corenlp\4-0-0\fr\stanford-corenlp-4.0.0-models-fr.jar to system classloader
2020-09-09 10:49:56,353 [CORENLP-0] ERROR CoreNlpNerModels - failed loading NER
java.io.IOException: Unable to open "edu\stanford\nlp\models\ner\french-wikiner-4class.crf.ser.gz" as class path, filename or URL
        at edu.stanford.nlp.io.IOUtils.getInputStreamFromURLOrClasspathOrFileSystem(IOUtils.java:481)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1505)
        at edu.stanford.nlp.ie.AbstractSequenceClassifier.loadClassifier(AbstractSequenceClassifier.java:1497)
        at edu.stanford.nlp.ie.crf.CRFClassifier.getClassifier(CRFClassifier.java:2888)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:75)
        at org.icij.datashare.text.nlp.corenlp.models.CoreNlpNerModels.loadModelFile(CoreNlpNerModels.java:39)
        at org.icij.datashare.text.nlp.AbstractModels.load(AbstractModels.java:59)
        at org.icij.datashare.text.nlp.AbstractModels.get(AbstractModels.java:46)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.processNerClassifier(CorenlpPipeline.java:196)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.process(CorenlpPipeline.java:91)
        at org.icij.datashare.nlp.NlpConsumer.findNamedEntities(NlpConsumer.java:73)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:43)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:20)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2020-09-09 10:49:56,354 [CORENLP-0] WARN  NlpConsumer - error in consumer main loop
java.lang.NullPointerException: null
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.processNerClassifier(CorenlpPipeline.java:197)
        at org.icij.datashare.text.nlp.corenlp.CorenlpPipeline.process(CorenlpPipeline.java:91)
        at org.icij.datashare.nlp.NlpConsumer.findNamedEntities(NlpConsumer.java:73)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:43)
        at org.icij.datashare.nlp.NlpConsumer.call(NlpConsumer.java:20)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
2020-09-09 10:49:56,355 [CORENLP-0] INFO  NlpConsumer - exiting main loop
2020-09-09 10:49:56,355 [pool-2-thread-3] INFO  NlpApp - exiting run
2020-09-09 10:50:02,849 [elasticsearch[jVsOze3][management][T#2]] WARN  DiskThresholdMonitor - flood stage disk watermark [95%] exceeded on [jVsOze3SRVS1utr1n8VuHg][jVsOze3][C:\Users\julien.martin\AppData\Roaming\Datashare\index\nodes\0] free: 4.6gb[1.9%], all indices on this node will be marked read-only
2020-09-09 10:50:32,867 [elasticsearch[jVsOze3][management][T#3]] WARN  DiskThresholdMonitor - flood stage disk watermark [95%] exceeded on [jVsOze3SRVS1utr1n8VuHg][jVsOze3][C:\Users\julien.martin\AppData\Roaming\Datashare\index\nodes\0] free: 4.6gb[1.9%], all indices on this node will be marked read-only
```

**Desktop (please complete the following information):**
 - OS: WIndows 10 Pro v1903
 - Browser: e.g. Chrome
 - Version: e.g. 7.8.0

**Additional context**
stanford-corenlp-4.0.0-models-en.jar file is downloaded and not corrupted.

**Comment by bamthomas** [comment]
reopening because of unstable models loading issues

**Comment by bamthomas** [comment]
It seems there are failures when downloading models. Sometimes it shows exception like this one : 
```
2022-04-12 16:14:24,541 [CORENLP-0] INFO  CoreNlpNerModels - downloading models for language SPANISH under dist/models/corenlp/4-0-0/es                                                                            
2022-04-12 16:14:29,349 [s3-transfer-manager-worker-2] INFO  ServiceUtils - Retry the download of object dist/models/corenlp/4-0-0/es/stanford-corenlp-4.0.0-models-es.jar (bucket datashare-nlp)                  
com.amazonaws.SdkClientException: Unable to store object contents to disk: Connection reset 
        at com.amazonaws.services.s3.internal.ServiceUtils.downloadToFile(ServiceUtils.java:313) 
        at com.amazonaws.services.s3.internal.ServiceUtils.downloadObjectToFile(ServiceUtils.java:270) 
        at com.amazonaws.services.s3.internal.ServiceUtils.retryableDownloadS3ObjectToFile(ServiceUtils.java:402) 
        at com.amazonaws.services.s3.AmazonS3Client.getObject(AmazonS3Client.java:1488) 
        at com.amazonaws.services.s3.transfer.internal.DownloadPartCallable.call(DownloadPartCallable.java:60) 
        at com.amazonaws.services.s3.transfer.internal.DownloadPartCallable.call(DownloadPartCallable.java:33) 
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) 
        at java.lang.Thread.run(Thread.java:748) 
Caused by: javax.net.ssl.SSLException: Connection reset 
        at sun.security.ssl.Alert.createSSLException(Alert.java:127) 
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:324) 
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:267) 
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:262) 
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:135) 
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1146) 
        at sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1116) 
        at sun.security.ssl.SSLSocketImpl.access$200(SSLSocketImpl.java:72) 
        at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:815) 
        at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:137) 
```

But other times, it seems to be fully downloaded but the md5 comparison is failing. 

```
2022-04-12 16:18:15,771 [CORENLP-0] DEBUG  RemoteFiles - remote {dist/models/corenlp/4-0-0/es/stanford-corenlp-4.0.0-models-es.jar=c9ee7cf952aad9de4c10f5ebe97aae9b-33} local {dist/models/corenlp/4-0-0/es/stanford-corenlp-4.0.0-models-es.jar=138c9f7c66b491a4fe2b45d90daadba5-33} is equal ? false    
```

Then after several attempts, it happens that a model stored locally has the same md5 from the remote : 

```
2022-04-12 16:17:33,879 [CORENLP-0] DEBUG  RemoteFiles - remote {dist/models/corenlp/4-0-0/fr/stanford-corenlp-4.0.0-models-fr.jar=cb7db295d0d96740eac9375349028227-33} local {dist/models/corenlp/4-0-0/fr/stanford-corenlp-4.0.0-models-fr.jar=cb7db295d0d96740eac9375349028227-33} is equal ? true 
```

Then all following attempts are going to succeed.

**Comment by bamthomas** [comment]
German models were consistently failing so I try to download them by hand (with AWS console), and install the file in the model folder. Then it worked for german.

Then I tried with spanish as well and it failed with a weird message : 

```
022-04-12 16:41:11,999 [CORENLP-0] INFO  RemoteFiles - remote {} local {dist/models/corenlp/4-0-0/de/stanford-corenlp-4.0.0-models-de.jar=6c2561cfb9a8af65bb21f8cc52d85abd-32} is equal ? false
```
It's as if the remote file is not existing. That should not happen. 

I observed also that downloading the file with the browser was significantly slower than with Java S3 SDK. That could be because the file is cut into chunks and the chunks downloading is parallelized.

**Comment by bamthomas** [comment]
We've seen some cases also where the file is downloaded, the size (in bytes) in exactly the same, when we do a `jar tvf` on it it succeds, but when we want to extract some files from it, it fails : 

```
$ jar xvf  datashare-app/dist/models/corenlp/4-0-0/en/stanford-corenlp-4.0.0-models-en.jar META-INF/MANIFEST.MF
java.util.zip.ZipException: ZipFile invalid LOC header (bad signature)
        at java.base/java.util.zip.ZipFile$ZipFileInputStream.initDataOffset(ZipFile.java:1008)
        at java.base/java.util.zip.ZipFile$ZipFileInputStream.read(ZipFile.java:1018)
        at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.fill(ZipFile.java:468)
        at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:159)
        at java.base/java.io.FilterInputStream.read(FilterInputStream.java:107)
        at jdk.jartool/sun.tools.jar.Main.copy(Main.java:1249)
        at jdk.jartool/sun.tools.jar.Main.copy(Main.java:1277)
        at jdk.jartool/sun.tools.jar.Main.extractFile(Main.java:1454)
        at jdk.jartool/sun.tools.jar.Main.extract(Main.java:1400)
        at jdk.jartool/sun.tools.jar.Main.run(Main.java:403)
        at jdk.jartool/sun.tools.jar.Main.main(Main.java:1681)
```

**Comment by pirhoo** [comment]
This was fixed in 10.2.2
